Cabal/Distribution/PackageDescription/Configuration.hs

```diff
@@ -351,18 +351,21 @@ overallDependencies enabled (TargetSet targets) = mconcat depss
 -- | Collect up the targets in a TargetSet of tagged targets, storing the
 -- dependencies as we go.
 flattenTaggedTargets :: TargetSet PDTagged -> (Maybe Library, [(UnqualComponentName, Component)])
-flattenTaggedTargets (TargetSet targets) = foldr untag (Nothing, []) targets
-  where
-    untag (_, Lib _) (Just _, _) = userBug "Only one library expected"
-    untag (_, Lib l) (Nothing, comps) = (Just l, comps)
-    untag (_, SubComp n c) (mb_lib, comps)
-        | any ((== n) . fst) comps =
-          userBug $ "There exist several components with the same name: '" ++ unUnqualComponentName n ++ "'"
-
-        | otherwise = (mb_lib, (n, c) : comps)
-
-    untag (_, PDNull) x = x  -- actually this should not happen, but let's be liberal
-
+flattenTaggedTargets (TargetSet targets) = foldr untag (Nothing, []) targets where
+  untag (depMap, pdTagged) accum = case (pdTagged, accum) of
+    (Lib _, (Just _, _)) -> userBug "Only one library expected"
+    (Lib l, (Nothing, comps)) -> (Just $ redoBD lensLibBD l, comps)
+    (SubComp n c, (mb_lib, comps))
+      | any ((== n) . fst) comps ->
+        userBug $ "There exist several components with the same name: '" ++ display n ++ "'"
+      | otherwise -> (mb_lib, (n, redoBD lensBuildInfo c) : comps)
+    (PDNull, x) -> x  -- actually this should not happen, but let's be liberal
+    where
+      redoBD :: ((BuildInfo -> BuildInfo) -> (a -> a)) -> (a -> a)
+      redoBD bd_lens = bd_lens $ \bi -> bi { targetBuildDepends = fromDepMap depMap }
+
+      lensLibBD :: (BuildInfo -> BuildInfo) -> (Library -> Library)
+      lensLibBD f = \l -> l { libBuildInfo = f $ libBuildInfo l }
```

Comment:
we have that lens. it's `over L.buildInfo`. We also have `targetBuildDepends` (which can apply directly on `Library`):

https://github.com/haskell/cabal/blob/736c6f421488050893eb6efa55c76e7ba17d346e/Cabal/Distribution/Types/BuildInfo/Lens.hs#L182

Reply:
BTW, it's not `Lens`, it's `Setter` ;)

Reply:
😲😆